packages/shared/src/variants/badukWithAbstractBoard/abstractBoard/TrihexagonalBoardHelper.ts

```diff
@@ -0,0 +1,76 @@
+import { Grid } from "../../../lib/grid";
+import { Vector2D } from "./Vector2D";
+import { Intersection } from "./intersection";
+
+export class TrihexagonalBoardHelper {
+  private readonly x_base_vector: Vector2D;
+  private readonly y_base_vector: Vector2D;
+
+  constructor() {
+    this.x_base_vector = new Vector2D(1, 0);
+    this.y_base_vector = new Vector2D(
+      Math.cos(Math.PI / 3),
+      Math.sin(Math.PI / 3),
+    );
+  }
+
+  CreateTrihexagonalBoard(size: number): Intersection[] {
+    const _size = size;
+
+    let x_shift_pattern = 0;
+    let y_shift_pattern = 0;
+
+    switch (_size % 3) {
+      case 0: {
+        x_shift_pattern = -(_size / 3) + 2;
+        y_shift_pattern = -(_size / 3);
+        break;
+      }
+      case 1: {
+        x_shift_pattern = -(_size - 1) / 3;
+        y_shift_pattern = -(_size - 1) / 3;
+        break;
+      }
+      case 2: {
+        x_shift_pattern = -((_size - 2) / 3) - 4;
+        y_shift_pattern = -((_size - 2) / 3) - 1;
+        break;
+      }
+    }
+
+    return new Grid<null>(_size, _size)
+      .fill(null)
+      .map<Intersection | null>((_value, index, _grid) =>
+        ((index.x + x_shift_pattern) * 4 - (index.y + y_shift_pattern)) % 7 ===
+          0 || index.x + index.y >= _size
+          ? null
+          : new Intersection(
+              this.x_base_vector
+                .Multiply(index.x)
+                .Add(this.y_base_vector.Multiply(index.y)),
+            ),
+      )
+      .map<Intersection | null>((intersection, index, grid) => {
+        if (!intersection) {
+          return null;
+        }
+
+        grid.at({ x: index.x + 1, y: index.y })?.ConnectTo(intersection, true);
+        grid.at({ x: index.x, y: index.y + 1 })?.ConnectTo(intersection, true);
+        if (index.x > 0) {
+          grid
+            .at({ x: index.x - 1, y: index.y + 1 })
+            ?.ConnectTo(intersection, true);
+        }
+
+        return intersection;
+      })
+      .to2DArray()
+      .flat()
+      .filter((value) => value !== null)
+      .map((intersection, index) => {
+        intersection!.Identifier = index;
+        return intersection!;
+      });
```

Comment:
Very minor nit: You can make the filter callback a [type guard](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) (`value is Intersection`), eliminating the need for the non-null assertion (`!`)

```suggestion
      .filter((value): value is Intersection => value !== null)
      .map((intersection, index) => {
        intersection.Identifier = index;
        return intersection;
      });
```

From [typescript-eslint/no-non-null-assertion](https://typescript-eslint.io/rules/no-non-null-assertion/):

> TypeScript's `!` non-null assertion operator asserts to the type system that an expression is non-nullable, as in not null or undefined. Using assertions to tell the type system new information is often a sign that code is not fully type-safe. It's generally better to structure program logic so that TypeScript understands when values may be nullable.

Reply:
Oh nice, I didn't know that could be used like this. I've changed it accordingly, thank you for the hint!